bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -1085,8 +1085,12 @@ private void addMergeCommitWarningComment() {
         pr.addComment(message);
     }
 
+    static String getJcheckName(PullRequest pr) {
+        return pr.repository().forge().name().equals("GitHub") ? "jcheck-" + pr.repository().name() + "-" + pr.id() : "jcheck";
+    }
+
```

Comment:
Why does the `GitLab merge request` not need to change such name? I think GitLab also needs.

This method is now put in class `CheckRun` and is used in both `CheckRun` and `CheckWorkItem`. It seems not a good design. We can move the method to the interface `PullRequest` with a default implementation, just like below. Then we can use `pr.getJcheckName()` in both `CheckRun` and `CheckWorkItem`.

```
// file PullRequest.java
    default String getJcheckName() {
        return "jcheck-" + repository().name() + "-" + id();
    }
```

Reply:
Because there is no concept of a "check" in GitLab. We just store the check results in a comment in the PR, so there is no reason to make the name of the check unique.

Until now I my opinion was that the name of the jcheck check was something only the pull request bot module needed to know about, so putting that on the `PullRequest` interface didn't seem right. It was an implementation detail. But now since we need to share this information with testinfo, it would make sense to push this logic into `PullRequest`. The default method should in that case return "jcheck" and `GitHubPullRequest` can have the special implementation needed only on GitHub.

Reply:
> Because there is no concept of a "check" in GitLab. We just store the check results in a comment in the PR, so there is no reason to make the name of the check unique.

Thanks for the explanation.

> Until now my opinion was that the name of the jcheck check was something only the pull request bot module needed to know about, so putting that on the `PullRequest` interface didn't seem right. It was an implementation detail. But now since we need to share this information with testinfo, it would make sense to push this logic into `PullRequest`. The default method should in that case return "jcheck" and `GitHubPullRequest` can have the special implementation needed only on GitHub.

I understand what you considered. The name is indeed only used in PR bot. So now I think the current code is also okay.